cylc/flow/scheduler.py

```diff
@@ -1187,28 +1191,45 @@ def run_event_handlers(self, event, reason):
             self.host, self.server.port))
 
     def process_task_pool(self):
-        """Process ALL TASKS whenever something has changed that might
-        require renegotiation of dependencies, etc"""
+        """Queue and release tasks, and submit task jobs.
+
+        The task queue manages references to task proxies in the task pool.
+
+        Newly released tasks are passed to job submission multiple times until
+        associated asynchronous host select, remote init, and remote install
+        processes are done.
+
+        """
         LOG.debug("BEGIN TASK PROCESSING")
         time0 = time()
         if self._get_events_conf(self.EVENT_INACTIVITY_TIMEOUT):
             self.set_suite_inactivity_timer()
+
+        # Forget tasks that are no longer preparing for job submission.
+        self.pre_submit_tasks = [
+            itask for itask in self.pre_submit_tasks if
+            itask.waiting_on_job_prep
+        ]
+
         if self.stop_mode is None and self.auto_restart_time is None:
-            itasks = self.pool.get_ready_tasks()
-            if itasks:
+            # Add newly released tasks to those still preparing.
+            from datetime import datetime
```

Comment:
I wonder if this is preventing `datetime` from being imported or not.

```bash
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ python
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from cylc.flow.scheduler import Scheduler
>>> import sys
>>> [mod for mod in sys.modules.keys() if 'datetime' in mod]
['_datetime', 'datetime', 'metomi.isodatetime', 'metomi.isodatetime.timezone', 'metomi.isodatetime.exceptions', 'metomi.isodatetime.parser_spec', 'metomi.isodatetime.dumpers', 'metomi.isodatetime.data', 'metomi.isodatetime.parsers', 'graphene.types.datetime']
>>> 
```

I think importing `Scheduler` is loading `datetime` too transitively? (I couldn't find which module is causing that to load). But that's not loaded by default.

```bash
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ python
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> [mod for mod in sys.modules.keys() if 'datetime' in mod]
[]
>>> 

```

Reply:
oops, good spotting - that's a debug left-over